service/gcs/prot/protocol.go

```diff
@@ -655,6 +663,12 @@ type NegotiateProtocolResponse struct {
 	Capabilities GcsCapabilities
 }
 
+// GetBase assigns `npr.MessageResponseBase` if `nil` and returns
+// `npr.MessageResponseBase`.
+func (npr *NegotiateProtocolResponse) GetBase() *MessageResponseBase {
```

Comment:
Shouldn't have to reimplement this for each type--Go will automatically expose the methods on any embedded values. However, you'll need to change the embedded type from a pointer to a value (which makes more sense anyway--the response base is not an optional part of the response, so it doesn't need to be a pointer).